packages/connection-manager/src/js/ConnectionQueue/ConnectionQueue.js

```diff
@@ -0,0 +1,91 @@
+import { List } from "immutable";
+import ConnectionQueueItem from "./ConnectionQueueItem";
+
+/**
+ * Connection List which holds ConnectionQueueItems which hold Connections :).
+ * It wraps Immutable.List because we cant extend it easily
+ * (see https://github.com/facebook/immutable-js/issues/301)
+ * Also is immutable itself.
+ */
+export default class ConnectionQueue {
+  /**
+   * Initializes an instance of ConnectionQueue
+   * @constructor
+   * @param {Set} connections – Set of Connections
+   */
+  constructor(connections = List()) {
+    if (!List.isList(connections)) {
+      throw new Error("Given list has to be an instance of List.");
+    }
+    Object.defineProperty(this, "connections", {
+      get: () => connections
+    });
+  }
+
+  /**
+   * Return current size of queue
+   * @return {integer} size of queue
+   */
+  get size() {
+    return this.connections.size;
+  }
+
+  /**
+   * returns head of Set - no removal
+   * @return {AbstractConnection} – head Connection
+   */
+  head() {
```

Comment:
Let's make sure we handle the edge cases in these methods to create a robust implementation. Handling the edge cases for a method should be done in the method otherwise it is hard to reason about and follow.